File: 算法/代码随想录算法PDF/01数组/26.删除有序数组中的重复项.cpp
/*
 * @lc app=leetcode.cn id=26 lang=cpp
 *
 * [26] 删除有序数组中的重复项
 */

// @lc code=start
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int slowIndex = 0;  // 慢指针指示插入单一数值位置
        int nowNum = nums[0];
        for (int fastIndex = 0; fastIndex < nums.size(); fastIndex++) {  // 快指针指示遍历末尾
            if (nums[fastIndex] != nowNum) {
                nums[slowIndex++] = nowNum;
                nowNum = nums[fastIndex];
            }
        }
        nums[slowIndex++] = nowNum;  // 最后一个存入
        return slowIndex;
    }
};
// @lc code=end

File: 算法/代码随想录算法PDF/00调试源代码/26.删除有序数组中的重复项.py
#
# @lc app=leetcode.cn id=26 lang=python3
#
# [26] 删除有序数组中的重复项
#

# @lc code=start
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        slowIndex = 0
        numNow = nums[0]
        for fastIndex in range(len(nums)):
            if (nums[fastIndex] != numNow):
                nums[slowIndex] = numNow
                slowIndex += 1
                numNow = nums[fastIndex]
        nums[slowIndex] = numNow
        return slowIndex + 1
# @lc code=end

